apps/.storybook/config.js

```diff
@@ -160,7 +163,11 @@ storybook.setAddon(addStoriesGroup);
 storybook.addDecorator(story => {
   var rendered = story();
   return (
-    <Centered>{rendered}</Centered>
+    <Provider store={store}>
+      <Centered>
+        {rendered}
+      </Centered>
+    </Provider>
```

Comment:
I actually don't like the idea of doing this for all stories - I think one of the strengths of storybook is working on pure display-only components.  I understand some of our components need providers because they include connected descendants, but I'd propose an approach that let stories opt-in to the decorator.

Here's my counter-proposal (I _think_ this will work): Let's build a small addon, similar to the `withExperiments` addon I built a while back, that lets a story file both opt-in to a wrapper Provider and specify the reducers it will use.  It might look something like this:

```jsx
storybook.setAddon({
  withReduxStore(reducers = {}) {
    this.addDecorator(story => {
      this.store = createStore(combineReducers(reducers));
      return (
        <Provider store={this.store}>
          {story()}
        </Provider>
      );
    });
  }
});
```

And here's how one might use it:
```jsx
export default storybook => storybook
  .storiesOf('ShareAllowedDialog', module)
  .withReduxStore({
    publishDialog: publishDialogReducer
  })
  .addStoryTable([
    // Story contents...
  ]);
```

This should let stories opt-in to having a redux store in context, allow them to specify which reducers they need, and I think even let individual stories reference `storybook.store` if they need to manually modify state before the story is rendered.

Thoughts?

Reply:
I like it 👍.  I'm not tied to the original design at all — this seems to be clearly better!  @Bjvanminnen what do you think?

Reply:
agreed